docs/adr/0016-automatic-aggregate-creation.md

```diff
@@ -0,0 +1,84 @@
+# 16. Automatic Aggregate Creation
+
+Date: 2020-11-02
+
+## Status
+
+Accepted
+
+- Amends [3. Aggregate Lifetime Control](0003-aggregate-lifetime-control.md)
+
+## Context
+
+After implementing several real engines and business domains we have found the
+rules governing how `AggregateCommandScope.Create()`, `Root()`, `RecordEvent()`
+and `Destroy()` can be called in relationship to one another tends to lead to
+overly cumbersome implementations.
+
+### Effect on handler implementations
+
+From the handler implementors perspective there are numerous subtle interactions
+between the aggregate instance's "state of existence" and the domain logic.
+
+Perhaps most egregiously is the requirement that an instance already exist
+before calling `Root()`. This means that the state of the aggregate root itself
+can not be used to determine which events should be recorded without first
+creating the instance. A less obvious problem is that when the domain logic is
+such that `RecordEvent()` is not called within the same scope as a call to
+create `Create()` the engine must panic.
+
+The net result of this is that the handler becomes equally concerned with the
+notion of whether the instance exists or not than it is with the actual business
+logic. The subtle interactions between the domain implementation (especially
+when located within methods of the aggregate root) and the handler itself are
+difficult to glean by reading the code, and hard to reason about.
+
+In summary, the semantics of `Create()` do not help the handler implementor to
+implement their domain logic in a clear and concise way.
+
+### Effect on engine implementations
+
+From the engine implementors perspective it seems that a non-trivial amount of
+validation logic needs to be implemented within the scope to verify that the
+application code is using the scope correctly, even though the engine does not
+really benefit from this requirements.
+
+For example, event sourced engine implementations tend to call
+`AggregateMessageHandler.New()` to construct a new instance before apply
```

Comment:
```suggestion
`AggregateMessageHandler.New()` to construct a new instance before applying
```

Reply:
Somehow this second change suggestion didn't get added to the first review response.